pyccel/ast/variable.py

```diff
@@ -831,3 +831,6 @@ def __eq__(self, other):
             return self.name == other.name and self.lhs == other.lhs
 
         return False
+
+    def __hash__(self):
+        return hash(type(self).__name__+ self.name)
```

Comment:
Shouldn't the `lhs` be in the hash?

Reply:
good point

Reply:
adding the lhs to the hash we should be 100% that whatever  the type of lhs  it should have an ```__str__``` method, or some that can identify it as its own hash method